test/sets.jl

```diff
@@ -212,6 +212,9 @@ u = unique([1,1,2])
 @test length(u) == 2
 @test unique(iseven, [5,1,8,9,3,4,10,7,2,6]) == [5,8]
 @test unique(n->n % 3, [5,1,8,9,3,4,10,7,2,6]) == [5,1,9]
+# issue 20105
+@test @inferred(unique(x for x in 1:1)) == [1]
+@test unique(x for x in Any[1,1.0])::Vector{Real} == [1]
```

Comment:
Moreso the second. Did the `::Vector{Real}` appear in the original? If so apologies, I missed it. Best!